joblib/numpy_pickle.py

```diff
@@ -118,6 +143,23 @@ def read_array(self, unpickler):
             # The array contained Python objects. We need to unpickle the data.
             array = pickle.load(unpickler.file_handle)
         else:
+            numpy_array_alignment_bytes = \
+                self.safe_get_numpy_array_alignment_bytes()
+            if numpy_array_alignment_bytes is not None:
+                try:
+                    current_pos = unpickler.file_handle.tell()
+                    alignment = current_pos % numpy_array_alignment_bytes
+                    if alignment != 0:
+                        padding_length = (
+                            numpy_array_alignment_bytes - alignment)
+                        unpickler.file_handle.seek(
+                            current_pos + padding_length)
+                except io.UnsupportedOperation as exc:
+                    raise RuntimeError(
+                        'Trying to read a joblib pickle with bytes aligned '
+                        'numpy arrays in a file handle '
+                        'that does not support .tell') from exc
```

Comment:
I wonder if this new exception will not be considered as a regression for people loading joblib pickle from custom stream objects that do no support tell. One could assume that they use the same file object implementation at read and write time so it might be that of a problem in practice but it could still happen to have asymmetric file object types at write and read time.

To avoid this problem, we could use the previously discussed single-byte prefix encoding of the padding length instead of using tell at read time when `numpy_array_alignment_bytes` is defined on the wrapper instance. What do you think?

Reply:
Done, I modified the test accordingly